src/assets/scripts/sections/product.js

```diff
@@ -79,6 +83,34 @@ export default register('product', {
         this.updateImages.bind(this),
       );
     }
+
+    this.initImageSwitch();
+  },
+
+  initImageSwitch() {
+    if (!$(selectors.productThumbs).length) {
+      return;
+    }
+
+    $(selectors.productThumbs)
+      .on('click', (evt) => {
+        evt.preventDefault();
+
+        const imageId = evt.currentTarget.dataset.thumbnailId;
```

Comment:
As much as I love this, for consistency purposes I feel we should use `data('thumbnail-id')` while we're still using jQuery in this file.